### PR TITLE
Fix OBB xywh-angle concatenation bug

### LINKED ISSUE #7401: Yolov8 OBB validation bug
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### YOLOv8 Component

_No response_

### Bug

1) In ultralytics/engine/model.py val function 'custom = {'rect': True} ' custom hard coded. If the model is trained with rect=False, it will affect visualization and mAP.
image

2) In ultralytics/models/yolo/obb/val.py  '_process_batch' function iou is calculated like
'' iou = batch_probiou(gt_bboxes, torch.cat([detections[:, :4], detections[:, -2:-1]], dim=-1)) ''
predicted detection is like [x,y,h,w, class, angle]
but this 'torch.cat([detections[:, :4], detections[:, -2:-1]], dim=-1)' makes predicted bbox like [x,y,h,w, class] and it calculates IOU wrong.
image

### Environment

_No response_

### Minimal Reproducible Example

_No response_

### Additional

_No response_

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

### PR DESCRIPTION
**Pull request by Laughing-q**
Thank you 🙏 for your contribution to [Ultralytics](https://ultralytics.com) 🚀! Your effort in enhancing our repositories is greatly appreciated. To streamline the process and assist us in integrating your Pull Request (PR) effectively, please follow these steps:

1. **Check for Existing Contributions**: Before submitting, kindly explore existing PRs to ensure your contribution is unique and complementary.

2. **Link Related Issues**: If your PR addresses an open issue, please link it in your submission. This helps us better understand the context and impact of your contribution.

3. **Elaborate Your Changes**: Clearly articulate the purpose of your PR. Whether it's a bug fix or a new feature, a detailed description aids in a smoother integration process.

4. **Ultralytics Contributor License Agreement (CLA)**: To uphold the quality and integrity of our project, we require all contributors to sign the CLA. Please confirm your agreement by commenting below:

    _I have read the CLA Document and I hereby sign the CLA_

For more detailed guidance and best practices on contributing, refer to our ✅ [Contributing Guide](https://docs.ultralytics.com/help/contributing). Your adherence to these guidelines ensures a faster and more effective review process.

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://github.com/ultralytics/actions)<sub>

### 🌟 Summary
Improved bounding box handling for rotated objects in object detection.

### 📊 Key Changes
- Modified tensor concatenation to include the correct orientation information for bounding boxes during validation.
- Adjusted the non-max suppression utility to ensure it uses the correct angle for rotated bounding box detection.

### 🎯 Purpose & Impact
- Ensures that the angle information used in object detection is accurate, leading to better performance when dealing with rotated objects. 🔄
- Could potentially improve the precision of object detection models in applications where objects are commonly rotated. 🎯
- Users utilizing Ultralytics models for object detection will benefit from more reliable detection outcomes. 👁‍🗨

### CODE REVIEW

**Comment by Laughing-q** [comment]
@glenn-jocher this also fixed nms for obb task, which means we didn't get a correct nms functionality before this PR. But no worries about our obb models, I re-evaluated our obb models, all the final test mAP remains unchanged. :)
But this indeed affected our internal mAP calculation and probably also affected the choice of best model during my training, I've already launched a new training to test it.

**Comment by glenn-jocher** [comment]
@Laughing-q ah ok got it! Merging now!
